Content.Shared/Implants/SharedImplanterSystem.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Linq;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Implants.Components;
+using Content.Shared.Popups;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+
+namespace Content.Shared.Implants;
+
+public abstract class SharedImplanterSystem : EntitySystem
+{
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    public const string ImplanterSlotId = "implanter_slot";
+    public const string ImplantSlotId = "implantcontainer";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ImplanterComponent, EntInsertedIntoContainerMessage>(OnEntInserted);
+
+    }
+
+    private void OnEntInserted(EntityUid uid, ImplanterComponent component, EntInsertedIntoContainerMessage args)
+    {
+        component.NumberOfEntities = args.Container.ContainedEntities.Count;
+        var implantData = EntityManager.GetComponent<MetaDataComponent>(args.Entity);
+        component.ImplantData = (implantData.EntityName, implantData.EntityDescription);
+    }
+
+    //Instantly implant something and add all necessary components and containers.
+    //Set to draw mode if not implant only
+    public void Implant(EntityUid implanter, EntityUid target, ImplanterComponent component)
+    {
+        if (!_container.TryGetContainer(implanter, ImplanterSlotId, out var implanterContainer))
+            return;
+
+        var implant = implanterContainer.ContainedEntities.FirstOrDefault();
+
+        if (!TryComp<SubdermalImplantComponent>(implant, out var implantComp))
+            return;
+
+        //If the target doesn't have the implanted component, add it.
+        if (!HasComp<ImplantedComponent>(target))
+            EnsureComp<ImplantedComponent>(target);
+
+        var implantContainer = _container.EnsureContainer<Container>(target, ImplantSlotId);
+        implanterContainer.Remove(implant);
+        component.NumberOfEntities = implanterContainer.ContainedEntities.Count;
+        implantComp.EntityUid = target;
+        implantContainer.OccludesLight = false;
```

Comment:
Does OccludesLight need updating every time? Seems better to just do it once in the yml.